services/spar/src/Spar/Scim.hs

```diff
@@ -184,5 +184,5 @@ server ::
 server conf =
   ScimSite
     { config = toServant $ Scim.configServer conf,
-      users = \authData -> toServant (Scim.userServer @tag authData)
+      users = toServant . Scim.userServer @tag
```

Comment:
I personally find the explicit arguments and anonymous functions sometimes make for more readable code. I.e. I would find it cool if writing the line `\authData -> ...` would remain allowed to write without CI barking at me in the future.
I don't mind there being code of the old version of the diff or code in the new version of the diff existing; I'm just not sure it's a good idea to always force the latter writing style. Just because hlint found out that one argument doesn't have to be explicitly written and the code still compiles (in a similar way, one could add a linter which deletes all comments - code would still compile without comments also, and there'd be fewer lines in the text file), it doesn't mean it's a useful thing to do in caring for the readability of the code for your coworkers.

There's more indirection involved when reading the code when you need to keep looking at types of only partially applied functions. Perhaps that's just me though.

Not sure what others think here?

Reply:
Ah missted that. Yes, I agree with you. I also think this point-free style (the "point" is the argument) that hlint sometimes suggests is harder to read sometimes, extreme example: try entering `\x -> f (g x) x` to `http://pointfree.io/`. I wouldn't want to enforce it.

Reply:
Wars have been fought over this stuff. There's the whole idea of [concatenative languages](https://en.wikipedia.org/wiki/Concatenative_programming_language), where all you get is pointfree style, since there are no variable bindings at all.

Personally, I agree with the general sentiment expressed by the above comments, but I suspect it's mostly a matter of what people are used to, and most people are not used to pointfree style, especially when taken to an extreme.

Sometimes, though, the fact that an expression doesn't refactor well to pointfree is a sign of something messy going on, like leakage in an abstraction, or something like that, and it can be worthwhile to reorganise things to make the expression more amenable to this kind of "concatenative" style.

But I agree that it shouldn't be a rule that we just enforce all the time.

Reply:
`Avoid lambdas` has been disabled and left it to the programmer to decide.